### PR TITLE
fix(card): update accessibility

### PR DESCRIPTION
**Pull request by ariellalgilmore**
### Related Ticket(s)

#6693 - card in card focus visibility
#6596 - feature section eyebrow text screen reader
#6281 - card static eyebrow text screen reader
#6512 - card group over verbose
#5504 - card in card eyebrow text screen reader
#6005 - card section simple boundaries
#4614 - card section images boundaries
#4714 - content block cards boundaries

### Description

Updated Card component to no longer wrap entire card in `<a>` the `<a>` only wraps around the footer, so that accessibility users can tab through cards and optionally use arrow keys to read the copy, header, or text.

### Changelog

**Changed**

- updated card and card footer components for accessibility use
- updated styles for focus states

### CODE REVIEW

File: packages/web-components/examples/codesandbox/components/card-group/cdn.html [hunk 1]
```diff
@@ -52,7 +52,7 @@
               <path
                 d="M11.8 2.8L10.8 3.8 16.2 9.3 1 9.3 1 10.7 16.2 10.7 10.8 16.2 11.8 17.2 19 10z"></path>
             </svg>
-          </dds-card-footer>
+          </dds-card-cta-footer>
```
Comment (jeffchew) [inline packages/web-components/examples/codesandbox/components/card-group/cdn.html:55]: @ariellalgilmore is this an API change or a correction in the example?

Comment (ariellalgilmore) [reply packages/web-components/examples/codesandbox/components/card-group/cdn.html:55]: correction in the example. card-group should always use `<dds-card-cta-footer>`. `<dds-card-group-item>` extends `DDSCardCTA` so the footer should match.

**Comment by jeffchew** [comment]
The alignment of the arrow is now the bottom right in link list, this is intentional? 
image

File: packages/web-components/examples/codesandbox/components/card-group/cdn.html (hunk 1 above)
Comment (jeffchew) [reply packages/web-components/examples/codesandbox/components/card-group/cdn.html:55]: Ok, we might want to note it in the next release to change to `dds-card-cta-footer` if this change breaks for anyone using `dds-card-footer` instead.

Comment (ariellalgilmore) [reply packages/web-components/examples/codesandbox/components/card-group/cdn.html:55]: yeah good idea!

**Comment by jeffchew** [comment]
@ariellalgilmore it looks like cards are unclickable unless it's changed to `dds-card-cta-footer`, wonder if we should add some messaging for this? 

https://ibmdotcom-web-components-html-test-upstream.s3.us-east.cloud-object-storage.appdomain.cloud/deploy-previews/6642/index.html

**Review by jeffchew — APPROVED** [review]
LGTM!

As discussed, we should include a PSA on the fixed documentation around card-group in our next release. Might even want adopters to know about it when the first rc comes out.

**Pull request merged**